### PR TITLE
M3-3201 Domains - Analytics

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

Adding domain creation analytics in a similar pattern to what John implemented here: https://github.com/linode/manager/pull/5190

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Currently we have 2 ways to create domains:
• From the 'Add New' menu dropdown
• From the domains landing page

Please test both these methods of Domain creation. Test that they work as expected, and that the appropriate GA events are being sent.

### CODE REVIEW

File: packages/manager/src/features/Domains/DomainDrawer.tsx [hunk 1]
```diff
@@ -611,6 +613,7 @@ class DomainDrawer extends React.Component<CombinedProps, State> {
               path(['ipv6'], selectedDefaultLinode)
             )
               .then(() => {
+                sendCreateDomainEvent(origin);
```
Comment (johnwcallahan) [inline packages/manager/src/features/Domains/DomainDrawer.tsx]: Could you call this function once, after the domain is created? (Maybe here: https://github.com/linode/manager/pull/5608/files#diff-2cdcc7824e5d54fc82beb80712dcc209R598)

File: packages/manager/src/features/Domains/DomainsLanding.tsx [hunk 2]
```diff
@@ -240,6 +241,12 @@ export class DomainsLanding extends React.Component<CombinedProps, State> {
     });
   };
 
+  openCreateDomainDrawer = (e: any) => {
```
Comment (johnwcallahan) [inline packages/manager/src/features/Domains/DomainsLanding.tsx]: You should be able to type `e` as `React.MouseEvent<HTMLElement>`.

File: packages/manager/src/features/Domains/DomainsLanding.tsx [hunk 3]
```diff
@@ -464,7 +471,7 @@ const EmptyCopy = () => (
 );
 
 const RenderEmpty: React.StatelessComponent<{
-  onClick: () => void;
+  onClick: (e: any) => void;
```
Comment (johnwcallahan) [inline packages/manager/src/features/Domains/DomainsLanding.tsx]: Same typing here.

**Comment by WilkinsKa1** [comment]
@johnwcallahan thanks for reviewing! I addressed your feedback.

**Pull request merged**